php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -1408,6 +1408,22 @@ public ExpressionTree memberExpression(ExpressionTree object, Optional<List<Expr
     return result;
   }
 
+  public ExpressionTree newObjectExpression(ExpressionTree object, Optional<List<ExpressionTree>> memberAccesses, Optional<FunctionCallTree> functionCall) {
+    ExpressionTree result = object;
+
+    for (ExpressionTree memberAccess : optionalList(memberAccesses)) {
+      if (memberAccess.is(Kind.OBJECT_MEMBER_ACCESS, Kind.CLASS_MEMBER_ACCESS)) {
+        result = ((MemberAccessTreeImpl) memberAccess).complete(result);
+      }
+    }
+
+    if (functionCall.isPresent()) {
+      result = ((FunctionCallTreeImpl) functionCall.get()).complete(result);
```

Comment:
I don't like those `complete` methods here and a few lines above.
I hope we can get rid of them one day.
Not in this PR.